Refactoring of British Raj and Forming Greater British Raj

John-Timothy opened the pull request:
British Raj consists of a custom region of Southern India and includes some of the Duchies from the Bengal Empire Greater British Raj on the other hand consists of all the regions of India

John-Timothy committed:
Refactoring of British Raj and Forming Greater British Raj

British Raj consists of a custom region of Southern India and includes some of the Duchies from the Bengal Empire
Greater British Raj on the other hand consists of all the regions of India

HansKalsi commented on common/decisions/00_major_decisions_british_india.txt:
```diff
@@ -30,6 +30,81 @@ form_british_raj_decision = {
+		NOT = {
+			has_title = title:e_british_raj
+		}
+	}
```
The is_shown for the greater raj decision should only be shown if the character has the British Raj title (you'll never wanna see both decisions at the same time)

Aka remove the NOT wrapper

HansKalsi replied:
You appear to have removed the entire NOT codeblock...
The NOT wrapper is not everything, it's the "wrapper", aka:
```
NOT = { # <-- This is the wrapper
   ...anything # <-- This is the content
}
```

You need the title content since you're basically saying, the character need to have the British Raj title for this decision to appear.

Your update has made it so any Britannic culture character will see this decision.

HansKalsi commented on localization/english/from_castles_to_bastilles_l_english.yml:
```diff
@@ -108,7 +108,14 @@
   form_british_raj_decision_desc: "The time has come for us to claim our Indian jewel"
-  form_british_raj_decision_tooltip: "Consolidate the Indian Empires into the British Raj"
-  form_british_raj_decision_effect_tooltip: "All Indian Empires assimilate under British Raj and you gain the British Raj Empire title"
+  form_british_raj_decision_tooltip: "Consolidate the Southern Indian Kingdoms into the British Raj"
```
I'd suggest referring to the region as 'mainland' instead of 'southern'

HansKalsi commented on common/decisions/00_major_decisions_british_india.txt:
```diff
@@ -30,6 +30,81 @@ form_british_raj_decision = {
+	is_valid = {
+		AND = {
+			completely_controls_region = custom_british_raj
+		}
```
Don't need the AND wrapper as it's just one item

John-Timothy committed:
Refactor: Adjustments based on code review comments (Removed NOT and AND conditions)

John-Timothy commented:
Pushed a newly refactored codes based on the suggested changes

John-Timothy committed:
Added checker to have title of British Raj before enabling the decision for Greater British Raj

HansKalsi commented on common/decisions/00_major_decisions_british_india.txt:
```diff
@@ -31,6 +31,7 @@ form_british_raj_decision = {
 	}
 
 	is_valid = {
+			has_title = title:e_british_raj
```
No, as before, we only need this in the `is_shown`.
Reasoning is we don't want to show this to everyone and have them not able to use it because they don't have the British Raj title.

Aka including only in `is_shown` means only someone who completed the first British Raj related decision will see this 'Greater British Raj' decision

HansKalsi replied:
My original comment said just to remove the wrapper, not also move the code

HansKalsi replied:
Also I just realised, this has been placed in the wrong decision.

John-Timothy committed:
Refactor: Adjustments to Has_Title based on suggestions on PR Review

HansKalsi committed:
- Updated game version to latest
- Added custom gfx for both decisions
- Fixed issues in localization and added further flavour
- Fixed incorrect custom_tooltip set (was using same for both decisions)
- Fixed custom british raj region to include all expected landmass
